pkgs/development/tools/aws-sam-cli/default.nix

```diff
@@ -105,12 +106,18 @@ python3.pkgs.buildPythonApplication rec {
     export PATH="$PATH:$out/bin:${lib.makeBinPath [ git ]}"
   '';
 
-  pytestFlagsArray = [
-    "tests"
-    # Disable warnings
-    "-W"
-    "ignore::DeprecationWarning"
-  ];
+  pytestFlagsArray =
+    [
+      "tests"
+      # Disable warnings
+      "-W"
+      "ignore::DeprecationWarning"
+    ]
+    ++ lib.optionals (stdenv.hostPlatform.isDarwin && stdenv.hostPlatform.isx86_64) [
+      # Disable warnings when falling back to kqueue
+      "-W"
+      "ignore::UserWarning:watchdog.observers"
+    ];
```

Comment:
Just drop https://github.com/aws/aws-sam-cli/blob/develop/pytest.ini#L7-L18 instead, we don't care about another extra warning

Reply:
@SuperSandro2000 
Thank you for your quick review!

How about replacing the `error` action in pytest.ini, which causes tests to fail when a warning is emitted, with the `default` action as follows?

https://github.com/blyoa/nixpkgs/commit/4ecf03f8d81164c18684d423cf2c218cc5594386

This change may be easier to implement than completely dropping `filterwarnings` from the pytest.ini. Also, the ignored warnings in original pytest.ini remain ignored, which might help in debugging future build errors from logs.

If it's better to completely drop `filterwarnings` as you suggested, I'll push the following commit:

https://github.com/blyoa/nixpkgs/commit/1e885a7bdaba1fb1e8bb03b778388c5810a3ad41

Reply:
@SuperSandro2000
Thanks again for your review.

I've provided two options based on your feedback:

*   Removing the `filterwarnings` section entirely (https://github.com/blyoa/nixpkgs/commit/1e885a7bdaba1fb1e8bb03b778388c5810a3ad41)

*   Modifying the warning behavior in `pytest.ini` (https://github.com/blyoa/nixpkgs/commit/4ecf03f8d81164c18684d423cf2c218cc5594386)

Just to confirm, would you prefer the first option, removing the `filterwarnings` section entirely (1e885a7), as per your initial suggestion?

I'll be decommissioning my x86_64-darwin machine in a few days, so I won't be able to easily do local checks after that.

Reply:
We don't want to maintain and rebase the patch file, so the sed it is https://github.com/blyoa/nixpkgs/commit/4ecf03f8d81164c18684d423cf2c218cc5594386

Reply:
@SuperSandro2000
Thank you for the reply!

I updated the PR based on https://github.com/blyoa/nixpkgs/commit/4ecf03f8d81164c18684d423cf2c218cc5594386 and rebased the head branch to resolve the conflict.